File: src/lgstar.pl
%Author: Michael Austin
% Academic Integrity Statement: I have complied with the WVU Academic
% Integrity policy and the submissions I make are solely my own. Etc.
% Etc.
lg2(0,0).
lg2(1,0).
lg2(N,Iterations) :-
    N > 1,
    Nsub is N // 2,
    lg2(Nsub, Iters1),
    Iterations is Iters1 + 1.

lgstar(N,0) :- N =< 1.
lgstar(N,Iterations) :-
    N > 1,
    lg2(N,Nlg),
    lgstar(Nlg,Iters2),
    Iterations is 1 + Iters2.

File: src/LgStar.java
/*
* Author: Michael Austin
* Academic Integrity Statement: I have not violated the WVU Academic Integrity Policy or cheated. The work below
* is solely my own. Etc. Etc.
*
* It wasn't explicitly stated that we could not use Math.log(), but it also wasn't stated that we could, so I just
* implemented everything in the Java-y version of my Prolog script. Ignored a change of base in the typical sense and
* just did the iterative division that a logarithm essentially is...
*
* Overloaded lgstar methods. Examples of calling in the Caller class, attached.
 */

public class LgStar {

    public static int lgbase2(int N){
        int result = -1;
        if (N > 1){
            while(N > 0){
                N /= 2;
                result += 1;
            }
        }else if(N < 0){
            return -1;
        }else{
            result = 0;
        }
        return result;
    }

    public static int lgstar(int N){
        int count = 0;
        int Iters = lgbase2(N);
        if(Iters == 0){
            return count;
        }
        else{
            while (Iters > 0){
                count++;
                Iters = lgbase2(Iters);

            }
        }
        return count;
    }

    static boolean lgstar(int N, int Iters){
        boolean res = false;
        int test = lgstar(N);
        if(Iters == test){
            res = true;
        }
        return res;
    }
}
